File: data-structure/js/unordered-array.js
var UnorderedArray = /** @class */ (function () {
    function UnorderedArray(length) {
        this.lastPos = -1;
        this.length = length;
        this.values = new Array(length);
    }
    UnorderedArray.prototype.print = function () {
        if (this.lastPos === -1)
            return console.log("The array is empty");
        for (var i = 0; i <= this.lastPos; i++) {
            console.log("".concat(i, " -- ").concat(this.values[i], " "));
        }
    };
    UnorderedArray.prototype.insert = function (value) {
        if (this.length === this.lastPos + 1)
            return console.error("Cannot insert, the array is already full");
        this.lastPos++;
        this.values[this.lastPos] = value;
    };
    UnorderedArray.prototype.find = function (value) {
        for (var pos in this.values) {
            if (this.values[pos] === value)
                return Number(pos);
        }
        return -1;
    };
    UnorderedArray.prototype.remove = function (value) {
        var pos = this.find(value);
        if (pos === -1)
            return pos;
        for (var i = pos; i <= this.lastPos; i++) {
            this.values[i] = this.values[i + 1];
        }
        this.lastPos--;
    };
    return UnorderedArray;
}());
var unorderedArray = new UnorderedArray(5);
unorderedArray.print(); // [ ]
unorderedArray.insert(1);
unorderedArray.insert(2);
unorderedArray.insert(3);
unorderedArray.insert(4);
unorderedArray.insert(5);
unorderedArray.print(); // [1, 2, 3, 4, 5]
unorderedArray.insert(6); // cannot insert
console.log(unorderedArray.find(7)); // -1
console.log(unorderedArray.find(3)); // 2
console.log(unorderedArray.remove(6)); // -1
unorderedArray.remove(3);
unorderedArray.insert(6);
unorderedArray.print(); // [1, 2, 4, 5, 6]

File: data-structure/java/UnorderedArray.java
public class UnorderedArray {
    private int capacity;
    private int lastPos = -1;
    private int values[];

    public UnorderedArray(int capacity) {
        this.capacity = capacity;
        this.values = new int[capacity];
    }

    private boolean empty() {
        return this.lastPos == -1;
    }

    private boolean filled() {
        return this.lastPos == this.capacity - 1;
    }

    public void print() {
        for (int i = 0; i <= this.lastPos; i++) {
            System.out.println(i + " -- " + this.values[i]);
        }
    }

    public void push(int value) {
        if (this.filled()) {
            System.out.println("The array is already filled");
            return;
        }

        this.lastPos += 1;
        this.values[this.lastPos] = value;
    }

    public void remove(int value) {
        if (this.empty()) {
            System.out.println("The array is empty");
            return;
        }

        int index = this.search(value);
        if (index == -1) {
            return;
        }

        while (index < this.lastPos) {
            this.values[index] = this.values[index + 1];
            index++;
        }
        this.lastPos -= 1;

    }

    public int search(int value) {
        if (this.empty()) {
            return -1;
        }

        int index = -1;

        for (int i = 0; i <= this.lastPos; i++) {
            if (this.values[i] == value) {
                index = i;
                break;
            }
        }

        return index;
    }

    public static void main(String[] args) {
        UnorderedArray u = new UnorderedArray(5);

        u.push(1);
        u.push(2);
        u.push(3);
        u.push(4);
        u.push(5);
        u.print();
        u.remove(1);
        u.remove(3);
        u.remove(5);
        u.push(0);
        u.print();
    }
}